crates/fluxus-transformers/src/transform_source.rs

```diff
@@ -31,11 +31,50 @@ impl<T: Clone + Send + Sync + 'static> Source<T> for TransformSource<T> {
     }
 
     async fn next(&mut self) -> StreamResult<Option<Record<T>>> {
-        let inner = Arc::clone(&self.inner);
-        unsafe {
-            // Safe because we have exclusive access through &mut self
-            let source = &mut *(Arc::as_ptr(&inner) as *mut InnerSource<T>);
-            source.next().await
+        loop {
+            let inner = Arc::clone(&self.inner);
+            let record = unsafe {
+                // Safe because we have exclusive access through &mut self
+                let source = &mut *(Arc::as_ptr(&inner) as *mut InnerSource<T>);
+                source.next().await?
+            };
+
+            match record {
+                Some(record) => {
+                    let mut current_record = Some(record);
+                    let mut filtered_out = false;
+
+                    // Apply each operator in sequence
+                    for op in &self.operators {
+                        if let Some(rec) = current_record {
+                            let operator = Arc::clone(op);
+                            let results = unsafe {
+                                // Safe because we have exclusive access through &mut self
+                                let op = &mut *(Arc::as_ptr(&operator) as *mut InnerOperator<T, T>);
+                                op.process(rec).await?
+                            };
+
+                            // If the operator filtered out the record (empty results), mark as filtered
+                            if results.is_empty() {
+                                filtered_out = true;
+                                current_record = None;
+                                break;
+                            } else {
+                                // Otherwise, take the first result for the next operator
+                                current_record = Some(results[0].clone());
```

Comment:
Only the first result was taken. If there are operators that produce multiple results in the future, there will be problems here. 
eg. `flat_map`

Reply:
Without lazy evaluation, I think there would be a buffer to store extra records.
pseudo code:
```rust
fn next(&mut self) -> Result<Record>{
    if self.buffer.len() > 0 {
        return Ok(self.buffer.pop().unwrap());
    }
    let next = self.inner.next();
    let mut records = vec![next];
    for op in self.operators.iter() {
        let mut cur_op_res = vec![];
        while let Some(record) = records.pop() {
            let out = op(record);
            cur_op_res.extend(out);     
        }
        records = cur_op_res;
    }
    self.buffer = records;
    self.buffer.pop()
}
```